mwdb/web/src/components/Shares.js

```diff
@@ -150,7 +160,10 @@ class SharesBox extends Component {
                 </thead>
                 <tbody>
                     {
-                        this.state.items.map((item, idx) => <ShareItem key={idx} {...item}/>)
+                        this.state.items
+                            .sort((a, b) => a.related_object_dhash > b.related_object_dhash)
+                            .sort((a) => a.related_object_dhash !== this.props.id)
```

Comment:
I don't really like this implementation. First of all: sort should return 1, 0 or 1. This probably will work ok if we assume that `Array.sort()` is stable and doesn't change the order of equal elements, but we can't really assume that since the stability requirement is from ES2019 standard.

https://stackoverflow.com/questions/3026281/what-is-the-stability-of-the-array-sort-method-in-different-browsers

```suggestion
                            .sort((a, b) => {
                                if(a.related_object_dhash !== b.related_object_dhash)
                                {
                                    // Current object should be on top
                                    if(b.related_object_dhash === this.props.id) return 1;
                                    if(a.related_object_dhash === this.props.id) return -1;
                                    // Inherited entries order by dhash
                                    if(a.related_object_dhash > b.related_object_dhash) return 1;
                                    if(a.related_object_dhash < b.related_object_dhash) return -1;
                                }
                                const a_time = Date.parse(a.access_time);
                                const b_time = Date.parse(b.access_time);
                                // The same dhash order by time
                                if(a_time > b_time) return 1;
                                if(a_time < b_time) return -1;
                                return 0;
                            })
```